util/eventbus/eventbustest/eventbustest.go

```diff
@@ -201,3 +201,40 @@ func eventFilter(f any) filter {
 		return fixup(fv.Call([]reflect.Value{args[0].Elem()}))
 	}).Interface().(filter)
 }
+
+// Injector holds a map with [eventbus.Publisher], tied to an [eventbus.Client]
+// for testing purposes.
+type Injector struct {
+	client     *eventbus.Client
+	publishers map[any]any
+}
+
+// NewInjector constructs an [Injector] that can be used to inject events into
+// the the stream of events used by code under test. After construction the
+// caller may use [Inject] to insert events into the bus.
+func NewInjector(t *testing.T, b *eventbus.Bus) *Injector {
+	inj := &Injector{
+		client:     b.Client(t.Name()),
+		publishers: make(map[any]any),
+	}
+	t.Cleanup(inj.client.Close)
+
+	return inj
+}
+
+// Inject inserts events of T onto an [eventbus.Bus]. If an [eventbus.Publisher]
```

Comment:
Maybe note in the comment that the publication is synchronous, so the caller can assume the publication is complete when `Inject` returns.